src/components/SchemaRow/useChoices.ts

```diff
@@ -0,0 +1,87 @@
+import { isReferenceNode, isRegularNode, SchemaNode } from '@stoplight/json-schema-tree';
+import { last } from 'lodash';
+import * as React from 'react';
+
+import { isComplexArray, isParentNode } from '../../tree';
+import { printName } from '../../utils';
+
+type Choice = {
+  title: string;
+  type: SchemaNode;
+};
+
+function calculateChoiceTitle(node: SchemaNode, isPlural: boolean): string {
+  const complexObjectSuffix = isPlural ? '-s' : '';
+  const primitiveSuffix = isPlural ? 's' : '';
+  if (isRegularNode(node)) {
+    const realName = printName(node);
+    if (realName) {
+      return realName + complexObjectSuffix;
+    }
+    return (node.primaryType ?? 'object') + primitiveSuffix;
+  }
+  if (isReferenceNode(node)) {
+    const lastPiece = last(node.value?.split('/') ?? []);
```

Comment:
You probably want to use `pointerToPath` to handle `~0` and ~1`.

Reply:
It's in `@stoplight/json`.

Reply:
Thank you, I'll do that